File: Estudos/semana-3/pilha/pilha.py
class Stack:
    def __init__(self):
        self.items = []

    # Adiciona um elemento ao topo da pilha
    def push(self, item):
        self.items.append(item)

    # Remove o elemento do topo da pilha
    def pop(self):
        if self.is_empty():
            return "Pilha vazia"
        return self.items.pop()

    # Retorna o elemento do topo da pilha
    def peek(self):
        return self.items[-1]

    # Verifica se a pilha está vazia
    def is_empty(self):
        return len(self.items) == 0

    # Retorna o tamanho da pilha
    def size(self):
        return len(self.items)

# Exemplo de uso
pilha = Stack()
pilha.push(10)
pilha.push(20)
print(pilha.pop())  # 20
print(pilha.peek())  # 10

File: Estudos/AlgortimosDeBusca/semana-3/pilha/pilha.js
class Stack {
    constructor() {
        this.items = [];
    }

    // Adiciona um elemento ao topo da pilha
    push(element) {
        this.items.push(element);
    }

    // Remove o elemento do topo da pilha
    pop() {
        if (this.items.length === 0) return "Pilha vazia";
        return this.items.pop();
    }

    // Retorna o elemento do topo da pilha
    peek() {
        return this.items[this.items.length - 1];
    }

    // Verifica se a pilha está vazia
    isEmpty() {
        return this.items.length === 0;
    }

    // Retorna o tamanho da pilha
    size() {
        return this.items.length;
    }
}

// Exemplo de uso
const pilha = new Stack();
pilha.push(10);
pilha.push(20);
console.log(pilha.pop()); // 20
console.log(pilha.peek()); // 10